### ISSUE TITLE
Login request is sent without port

### ISSUE THREAD

**Issue by Konpon96** [rancher/rancher#26422]
**What kind of request is this (question/bug/enhancement/feature request):** Bug

**Steps to reproduce (least amount of steps as possible):**

1. Deploy Rancher 2.3.5 Helm Chart with Rancher-generated certificates with the Ingress exposing ports different than 80/443 for respectively HTTP and HTTPS (in my case that was 8080 and 8443)
2. After opening the Rancher panel site for the first time, configure the hostname to the same one as from Helm Chart values, with added port (in my case `https://rancher.rancher:8443`)
3. Wait for the local cluster to be provisioned
2. Upgrade the Rancher Chart from version 2.3.5 to 2.4.1
3. Try to login

**Result:**

The login request made after clicking "login" button on the Rancher's panel login site doesn't include the port on which Rancher is actually deployed which makes it impossible to login. After rollbacking Rancher Chart to version 2.3.5 the login button works properly.

**Other details that may be helpful:**

As you can see in the screenshot from Chrome DevTools which I attached below, the request made after clicking "login" button doesn't include the port, which is present in the browser address field:

![Screenshot from 2020-04-01 15-39-29](https://user-images.githubusercontent.com/4958634/78144859-75b0e180-7430-11ea-9a67-30287a7c3151.png)

![Screenshot from 2020-04-01 15-40-01](https://user-images.githubusercontent.com/4958634/78145889-be1ccf00-7431-11ea-8e3e-771fc4eda5a9.png)

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.4.1
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Deployed with RKE with Rancher installed from Helm Chart (3 control plane machines, 3 workers)
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VM (2 CPU, 4 GB RAM for control plane, 4 CPU, 8 GB RAM for workers)
- Kubernetes version (use `kubectl version`): 1.17.2

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:20:10Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:22:30Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea
 Built:             Wed Nov 13 07:25:41 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea
  Built:            Wed Nov 13 07:24:18 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

```

**Comment by ibuildthecloud** [comment]
I'm confused about your setup.  I think I know the fix, but regarding your setup I don't understand it.  Can you share the specific helm values you use to install?  I don't understand how ingress can be :8443 but you access the site through :443.  I would think ingress will drop the request if you don't configure the host as :443.

**Comment by toddexp** [comment]
I believe we are facing a similar issue. We also upgraded from 2.3.5 to 2.4.1 and can not login. We are using [external tls termination](https://rancher.com/docs/rancher/v2.x/en/installation/options/chart-options/#external-tls-termination) though. When I inspect our headers the requested url is adding port 80. 

![rancher-error](https://user-images.githubusercontent.com/45802865/78176686-ec4ae080-742a-11ea-91ff-232656a1c3c0.png)

**Comment by bmdepesa** [comment]
@toddexp how did you do your initial Rancher installation w/ RKE and helm?

**Comment by toddexp** [comment]
@bmdepesa Yes. This is a Rancher HA install with RKE and helm.

**Comment by bmdepesa** [comment]
@toddexp can you elaborate on your Helm set values and any ingress options you supplied in the RKE yaml?

**Comment by toddexp** [comment]
@bmdepesa

RKE ingress section: `ingress:
  provider: nginx
  options:
    use-forwarded-headers: 'true'`

helm: `helm install rancher rancher-latest/rancher --namespace cattle-system --set hostname=example.com,rancherImageTag=v2.4.1,tls=external`

**Comment by khushboo-rancher** [comment]
Tested in v2.4.2-rc2, Worked fine when upgraded from v2.3.5 to v2.4.2-rc2

Steps:
1. I had a setup v2.4.1 HA set up with ingress pointing to 8080/8443 (3 nodes all roles)
2. Downgraded HA setup to v2.3.5
3. Upgraded it to v2.4.2-rc2
4. Login worked.

<img width="658" alt="Screen Shot 2020-04-01 at 1 40 26 PM" src="https://user-images.githubusercontent.com/60111667/78184475-6c1e7e00-741e-11ea-8a39-59dc3b961f6c.png">

I'll try with a setup initially with v2.3.5 and update the result.

**Comment by bmdepesa** [comment]
@toddexp I am not able to reproduce the same behavior. 

* v1.17.2 local cluster (RKE v1.04) (w/ forward headers)
* Rancher v2.3.5 HA w/ external TLS
* TLS offloaded to nginx lb

```
helm install rancher rancher-stable/rancher --version 2.3.5 --namespace cattle-system --set hostname=host.com --set tls=external

helm upgrade rancher rancher-latest/rancher --version 2.4.1 --namespace cattle-system --set hostname=host.com --set tls=external
```

After upgrading Rancher, I am able to login (including in new browser / session). I do not see `:80` appended to the hostname in the request.

**Comment by sangeethah** [comment]
@toddexp Could you please log a separate issue to track the issue you are seeing ? Issue that is noted in this bug originally is fixed `v2.4.2-rc2` and validated - https://github.com/rancher/rancher/issues/26422#issuecomment-607479384

**Comment by deniseschannon** [comment]
We have reverted the PR in the v2.4 line so that this login works as expected. An official fix will need to be created in v2.4 which is being tracked #26434.

**Comment by toddexp** [comment]
This 2.4.2 release fixed my issue as well. I am now able to login. Thank you all.
